modules/userdirectory/src/main/java/org/opencastproject/userdirectory/LtiUserProvider.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.userdirectory;
+
+import org.opencastproject.security.api.JaxbOrganization;
+import org.opencastproject.security.api.JaxbRole;
+import org.opencastproject.security.api.JaxbUser;
+import org.opencastproject.security.api.User;
+import org.opencastproject.security.api.UserProvider;
+
+import org.osgi.service.cm.ManagedService;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Iterator;
+
+public class LtiUserProvider implements UserProvider, ManagedService {
+  public static final String PROVIDER_NAME = "lti";
+
+  @Override
+  public String getName() {
+    return PROVIDER_NAME;
+  }
+
+  @Override
+  public Iterator<User> getUsers() {
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    return Collections.emptyIterator();
+  }
+
+  @Override
+  public User loadUser(String userName) {
+    if (!userName.startsWith("lti:")) {
```

Comment:
As I have already [pointed out](https://groups.google.com/a/opencast.org/d/msg/dev/YX_uCIUxfLA/m1p50M_iBgAJ), LTI can be [configured](https://docs.opencast.org/r/7.x/admin/modules/ltimodule/#configure-lti-optional) to trust the LTI provider. In this case, temporary LTI user names **do not** start with `lti:`! Depending on the LMS, the user name could be an ID or the actual user name in the LMS, as e.g. in Moodle.

Would it be possible to give users a special LTI role and differentiate based on that role?

Reply:
I understand what you're saying, and a solution emplying a user provider for the LMS, plus highly trusted users is far better than this `LtiUserProvider` thingy. We'll talk to our customer and if they agree, I'll delete this provider from the PR.
I'm not sure where you'd like to differentiate via a role, though. Can you elaborate?

Reply:
Now that I think about it, this won't work. I thought of adding a special role during the execution of `LtiLaunchAuthenticationHandler`, but this won't help you in a user provider that only gives you a name 🤦‍♂.